File: Backend/Challenge.Domain/Entities/PokemonStatus.cs
using Challenge.Domain.Aggregates;
using Challenge.Domain.Interfaces.Entities;
using Challenge.Domain.Interfaces.ValueObject;
using System.Text.Json.Serialization;

namespace Challenge.Domain.Entities
{
    public class PokemonStatus : IPokemonStatus
    {
        public int Id { get; set; }

        public bool IsDefault { get; set; }
        [JsonPropertyName("location_area_encounters")]
        public string? LocationAreaEncounters { get; set; }

        public string? Name { get; set; }

        public int Order { get; set; }

        public float Height { get; set; }

        public float Weight { get; set; }
        [JsonPropertyName("base_experience")]
        public int BaseExperience { get; set; }

        public List<PokemonMove> Moves { get; set; } = new List<PokemonMove>();

        public List<Stats> Stats { get; set; } = new List<Stats>();

        public List<Types> Types { get; set; } = new List<Types>();

        public List<Abilities> Abilities { get; set; } = new List<Abilities>();

        public List<Form> Forms { get; set; } = new List<Form>();

        [JsonPropertyName("game_indices")]
        public List<GameIndice> GameIndices { get; set; } = new List<GameIndice>();

        public Species Species { get; set; } = new Species();

        public PokemonStatus()
        {

        }
    }
}

File: JsonFiles/api/models/PokemonStatus.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Abilities } from './Abilities';
import type { Form } from './Form';
import type { GameIndice } from './GameIndice';
import type { PokemonMove } from './PokemonMove';
import type { Species } from './Species';
import type { Stats } from './Stats';
import type { Types } from './Types';

export type PokemonStatus = {
    id?: number;
    isDefault?: boolean;
    location_area_encounters?: string | null;
    name?: string | null;
    order?: number;
    height?: number;
    weight?: number;
    base_experience?: number;
    moves?: Array<PokemonMove> | null;
    stats?: Array<Stats> | null;
    types?: Array<Types> | null;
    abilities?: Array<Abilities> | null;
    forms?: Array<Form> | null;
    game_indices?: Array<GameIndice> | null;
    species?: Species;
};
